app/src/main/java/school/cactus/cactusnotesmvvmsample/SignupViewModel.kt

```diff
@@ -18,8 +20,11 @@ class SignupViewModel : ViewModel() {
     private val passwordValidator = PasswordValidator()
 
     fun onSignupButtonClick() {
-        emailError.value = emailValidator.validate(email.value)
-        usernameError.value = usernameValidator.validate(username.value)
-        passwordError.value = passwordValidator.validate(password.value)
+
+
+        emailError.value = try { context.getString(emailValidator.validate(email.value)) } catch (e: Resources.NotFoundException) { null }
+        usernameError.value = try { context.getString(usernameValidator.validate(username.value))} catch (e: Resources.NotFoundException) { null }
+        passwordError.value = try { context.getString(passwordValidator.validate(password.value)) } catch (e: Resources.NotFoundException) { null }
```

Comment:
Also I believe, we can let the `View` layer do the getString conversion since this function is tightly coupled with context.